items.cpp

```diff
@@ -784,6 +804,53 @@ AString *ItemDescription(int item, int full)
 		*temp += "silver as treasure.";
 	}
 
+	if(ItemDefs[item].flags & ItemType::MANPRODUCE) {
+		*temp += " This is a free-moving-item (FMI).";
+		MonType *mp = FindMonster(ItemDefs[item].abr,
+				(ItemDefs[item].type & IT_ILLUSION));
+		*temp += AString(" This FMI attacks with a combat skill of ") +
+			mp->attackLevel + ".";
+		for (int c = 0; c < NUM_ATTACK_TYPES; c++) {
+			*temp += AString(" ") + FMIResist(c,mp->defense[c], full);
+		}
+		if (mp->special && mp->special != NULL) {
+			*temp += AString(" ") +
+				"FMI can cast " +
+				ShowSpecial(mp->special, mp->specialLevel, 1, 0);
+		}
+		if (full) {
+			int hits = mp->hits;
+			int atts = mp->numAttacks;
+			int regen = mp->regen;
+			if (!hits) hits = 1;
+			if (!atts) atts = 1;
+			*temp += AString(" This FMI has ") + atts + " melee " +
+				((atts > 1)?"attacks":"attack") + " per round and takes " +
+				hits + " " + ((hits > 1)?"hits":"hit") + " to kill.";
+			if (regen > 0) {
+				*temp += AString(" This FMI regenerates ") + regen +
+					" hits per round of battle.";
+			}
+			*temp += AString(" This FMI has a tactics score of ") +
+				mp->tactics + ", a stealth score of " + mp->stealth +
+				", and an observation score of " + mp->obs + ".";
+		}
+
+		if (mp->spoiltype != -1) {
```

Comment:
Why free moving items have spoils?
As a something players produce we might not want it to drop anything right?

Reply:
Will be there spoils or not depends on item definition. In the sample catapult and steel defender `spoiltype` is set to have 0 spoils.